integration/kubernetes/helm-chart/alluxio/templates/_helpers.tpl

```diff
@@ -300,70 +300,6 @@ resources:
   {{- end }}
 {{- end -}}
 
-{{- define "alluxio.master.readinessProbe" -}}
-readinessProbe:
-  tcpSocket:
-    port: {{ index . "port" }}
-{{- end -}}
-
-{{- define "alluxio.jobMaster.readinessProbe" -}}
-readinessProbe:
-  tcpSocket:
-    port: {{ index . "port" }}
-{{- end -}}
-
-{{- define "alluxio.worker.readinessProbe" -}}
-readinessProbe:
-  tcpSocket:
-    port: rpc
-{{- end -}}
-
-{{- define "alluxio.jobWorker.readinessProbe" -}}
-readinessProbe:
-  tcpSocket:
-    port: job-rpc
-{{- end -}}
-
-{{- define "alluxio.master.livenessProbe" -}}
-livenessProbe:
-  tcpSocket:
-    port: {{ index . "port" }}
-  initialDelaySeconds: 15
-  periodSeconds: 30
-  timeoutSeconds: 5
-  failureThreshold: 2
-{{- end -}}
-
-{{- define "alluxio.jobMaster.livenessProbe" -}}
-livenessProbe:
-  tcpSocket:
-    port: {{ index . "port" }}
-  initialDelaySeconds: 15
-  periodSeconds: 30
-  timeoutSeconds: 5
-  failureThreshold: 2
-{{- end -}}
-
-{{- define "alluxio.worker.livenessProbe" -}}
-livenessProbe:
-  tcpSocket:
-    port: rpc
-  initialDelaySeconds: 15
-  periodSeconds: 30
-  timeoutSeconds: 5
-  failureThreshold: 2
-{{- end -}}
-
-{{- define "alluxio.jobWorker.livenessProbe" -}}
-livenessProbe:
-  tcpSocket:
-    port: job-rpc
-  initialDelaySeconds: 15
-  periodSeconds: 30
-  timeoutSeconds: 5
-  failureThreshold: 2
-{{- end -}}
-
```

Comment:
IMO the "main" yaml template (`statefulset.yaml` and `daemonset.yaml`) are already too long to read, so I'd prefer to keep that as short as possible but moving blocks into the `_helpers.tpl`. What do you guys think?

Reply:
`_helpers.tpl` serves the below purpose

1) if there is a function that can be reused at more than one place, then add that function to `_helpers.tpl` and use that function at multiple places. 
2) if there is any complex logic required, like looping and replacing some stuff etc. then it is better to add that in `_helpers.tpl` so that will abstract the overall complexity of the code from the user in `yaml` files.

The above probe use case does not fall in any of the above categories. Neither it has a function to be reused, nor it has any logic to add. 

If you still add that stuff in `_helpers.tpl` then you are adding more line of code Function declaration + Code (mostly same code and number of lines you would have in `yaml`) + Function usage (at only one place).

So, in my opinion, adding those blocks in `_helpers.yaml` does not make any sense.

Reply:
I'd agree with @nirav-chotai on this; the YAML templates are more readable without the abstraction of calling a function for the probe definitions given how straight-forward the are right now. I think it's fine to leave it out of `_helpers.tpl`.